src/v/cluster/topics_frontend.cc

```diff
@@ -1313,4 +1315,69 @@ ss::future<std::error_code> topics_frontend::decrease_replication_factor(
       _stm, _features, _as, std::move(cmd), timeout);
 }
 
+namespace {
+bool in_broker_list(
+  const model::broker_shard& bshard,
+  const std::vector<model::broker_shard>& list) {
+    return std::find(list.begin(), list.end(), bshard) != list.end();
+}
+} // namespace
+
+ss::future<partition_reassignments>
+topics_frontend::list_partition_reassignments(
+  model::ntp ntp, model::timeout_clock::time_point tout) {
+    auto result = co_await stm_linearizable_barrier(tout);
+    if (!result) {
+        co_return partition_reassignments{
+          .error_code = kafka::error_code::unknown_server_error,
+          .error_message = ssx::sformat(
+            "Unknown issue listing reassignments - {}", ntp)};
+    }
+
+    if (_topics.local().is_update_in_progress(ntp)) {
+        co_return partition_reassignments{
+          .error_code = kafka::error_code::replica_not_available,
+          .error_message = ssx::sformat("Update in progress - {}", ntp)};
+    }
+
+    // Current replica set is in the current partition assignment
+    auto current_assignment = _topics.local().get_partition_assignment(ntp);
+    if (!current_assignment.has_value()) {
+        co_return partition_reassignments{
+          .error_code = kafka::error_code::replica_not_available,
+          .error_message = ssx::sformat("No replica set - {}", ntp)};
+    }
```

Comment:
I need to take another look at the meanings for certain errors. Perhaps there is a more specific error than `unknown_server_error` for a failed result and `replica_not_available` may not be suitable for in progress updates